src/librustc/ty/constness.rs

```diff
@@ -63,13 +64,63 @@ impl<'tcx> TyCtxt<'tcx> {
 
 
 pub fn provide(providers: &mut Providers<'_>) {
-    /// only checks whether the function has a `const` modifier
+    /// Const evaluability whitelist is here to check evaluability at the
+    /// top level beforehand.
+    fn is_const_intrinsic(tcx: TyCtxt<'_>, def_id: DefId) -> bool {
+        match tcx.fn_sig(def_id).abi() {
+            Abi::RustIntrinsic |
+            Abi::PlatformIntrinsic => {
+                match tcx.item_name(def_id) {
+                    | sym::size_of
+                    | sym::min_align_of
+                    | sym::needs_drop
+                    | sym::type_id
+                    | sym::bswap
+                    | sym::bitreverse
+                    | sym::ctpop
+                    | sym::cttz
+                    | sym::cttz_nonzero
+                    | sym::ctlz
+                    | sym::ctlz_nonzero
+                    | sym::overflowing_add
+                    | sym::overflowing_sub
+                    | sym::overflowing_mul
```

Comment:
None of these seem to be implemented in `interpret/intrinsics.rs`?

Reply:
the intrinsic is named differently from the method on the integer types. I removed these duplicate entries